File: Submissions/Educational_065/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int t;
  cin >> t;

  for (int i = 0; i < t; i++) {
    int n;
    cin >> n;
    string s;
    cin >> s;
    string answer = "NO";
    for (int j = 0; j < n; j++) {
      if (s[j] == '8' && j + 10 < n) {
        answer = "YES";
        break;
      }
    }
    cout << answer << endl;
  }

}

File: Submissions/Educational_065/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int t = sc.nextInt();

    for (int i = 0; i < t; i++) {
      int n = sc.nextInt();
      String S = sc.next();
      String[] s = S.split("");
      String answer = "NO";
      for (int j = 0; j < n; j++) {
        if (s[j].equals("8") && j + 10 < n) {
          answer = "YES";
          break;
        }
      }
      System.out.println(answer);
    }

    sc.close();

  }

}